### PR TITLE
v0.30: filtering and NaN/infinity values

### LINKED ISSUE #2001: v0.30: Filtering and `NaN`/`Infinity` values
According to this issue on core, users cannot filter on `NaN` and `infinity` values. 

Actual behaviour with v0.30 is not totally clear yet: what should happen when filter includes a document field with one of these values? Do I read the PR correctly and `Infinity` is converted into a string?

### References
- SME: @Kerollmops 
- issue on core: https://github.com/meilisearch/meilisearch/issues/3000#issuecomment-1297197469
- PR on core: https://github.com/meilisearch/milli/pull/689

### PR DESCRIPTION
**Pull request by maryamsulemani97**
closes #2001

### CODE REVIEW

**Review by Kerollmops — APPROVED** [review]
Looks good to me!

I hope they understand that it will work perfectly and this is not a limitation on our side: it is not possible to write `NaN` or `inf` numbers in JSON, therefore it is impossible to store those as numbers in the database anyway! (ok you can use CSV but we handle them in the same way as in JSON).

**Comment by maryamsulemani97** [comment]
@Kerollmops 
Could you please share a link where this is explained:
>it is not possible to write `NaN` or `inf` numbers in JSON

I couldn't find anything 😕 
I can add it to clarify further that it's not a limitation on our side

File: learn/advanced/filtering_and_faceted_search.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ Updating `filterableAttributes` requires recreating the entire index. This may t
 By default, `filterableAttributes` is empty. This means that filters do not work without first explicitly adding attributes to the `filterableAttributes` list.
 :::
 
-Filters work with numeric and string values. Empty fields or fields containing an empty array will be ignored.
+Filters work with numeric and string values. Empty fields or fields containing an empty array will be ignored. Filtering with `inf` or `NaN` as numbers will throw an error as they are [not supported by JSON](https://en.wikipedia.org/wiki/JSON#Data_types). Filtering with  `inf` or `NaN` as strings will work for all fields except [geo fields](/learn/advanced/geosearch.md#preparing-documents-for-location-based-search).
```
Comment (guimachiavelli) [inline learn/advanced/filtering_and_faceted_search.md:21]: I'm not sure about `inf`—it's fairly obvious, I guess, but are all users going to understand it means `Infinity`?  Perhaps we can use `Infinity` or even just "infinite numbers"?

**Review by guimachiavelli — CHANGES_REQUESTED** [review]
I think it's good, just wondering if `inf` is clear enough

**Review by guimachiavelli — APPROVED** [review]
🎤

**Comment by maryamsulemani97** [comment]
bors merge

**Comment by maryamsulemani97** [comment]
bors merge

**Pull request merged**